### PR TITLE
Add hex input and validation to color picker.

### PR DESCRIPTION
**Pull request by GreenLightning**
As the title says. Please have a look at the code and tell me what you think.

### CODE REVIEW

File: src/core/picker.cpp [hunk 1]
```diff
@@ -82,16 +82,38 @@ void Picker::Show(PickerInfo* Picker, Color* Colors, uint32 BlankTexture) // TOD
...
-        col[2] = (int)(Picker->NewColor.b * 255.0f);
-        if (ImGui::InputInt3("RGB", col))
+        int rgb_color[3];
+        rgb_color[0] = (int)(Picker->NewColor.r * 255.0f);
+        rgb_color[1] = (int)(Picker->NewColor.g * 255.0f);
+        rgb_color[2] = (int)(Picker->NewColor.b * 255.0f);
+        if (ImGui::InputInt3("RGB", rgb_color))
         {
-            Picker->NewColor = Color(col[0], col[1], col[2]); // TODO: Clamping
+			int r = Math::Clamp(rgb_color[0], 0, 255);
```
Comment (ApoorvaJ) [inline src/core/picker.cpp]: Spaces, not tabs, please. :)

There are more lines below with the same problem.

File: src/core/picker.cpp [hunk 2]
```diff
@@ -82,16 +82,38 @@ void Picker::Show(PickerInfo* Picker, Color* Colors, uint32 BlankTexture) // TOD
...
+			int g = Math::Clamp(rgb_color[1], 0, 255);
+			int b = Math::Clamp(rgb_color[2], 0, 255);
+            Picker->NewColor = Color(r, g, b);
             Math::RGBtoHSV(Picker->NewColor.r, Picker->NewColor.g, Picker->NewColor.b,
                     Picker->CursorH, Picker->CursorSV.x, Picker->CursorSV.y);
         }
+		char hex_color[6 + 1]; // null-terminated
```
Comment (ApoorvaJ) [inline src/core/picker.cpp]: According to local variable convention, this variable should be named hexColor.

(I know I'm deviating from the convention myself in a few places, especially in terms of the capitalization, but I'll be whipping the code into shape soon.)

File: src/core/picker.cpp [hunk 3]
```diff
@@ -82,16 +82,38 @@ void Picker::Show(PickerInfo* Picker, Color* Colors, uint32 BlankTexture) // TOD
...
+		snprintf(hex_color, sizeof(hex_color) / sizeof(hex_color[0]), "%02x%02x%02x",
```
Comment (ApoorvaJ) [inline src/core/picker.cpp]: Why the division by sizeof(hex_color[0])?

File: src/core/picker.cpp [hunk 4]
```diff
@@ -82,16 +82,38 @@ void Picker::Show(PickerInfo* Picker, Color* Colors, uint32 BlankTexture) // TOD
...
+			(int)(Picker->NewColor.r * 255.0f),
+			(int)(Picker->NewColor.g * 255.0f),
+			(int)(Picker->NewColor.b * 255.0f));
```
Comment (ApoorvaJ) [inline src/core/picker.cpp]: In general, all of these `int`s should be `int32`s, but that is my fault as well. I haven't gotten around to cleaning up all the ints in my code. This is just an fyi for you. :)

File: src/core/picker.cpp [hunk 5]
```diff
@@ -82,16 +82,38 @@ void Picker::Show(PickerInfo* Picker, Color* Colors, uint32 BlankTexture) // TOD
...
+            Picker->NewColor = Color(r, g, b);
             Math::RGBtoHSV(Picker->NewColor.r, Picker->NewColor.g, Picker->NewColor.b,
                     Picker->CursorH, Picker->CursorSV.x, Picker->CursorSV.y);
         }
+        char hex_color[6 + 1]; // null-terminated
+        snprintf(hex_color, sizeof(hex_color) / sizeof(hex_color[0]), "%02x%02x%02x",
+            (int)(Picker->NewColor.r * 255.0f),
+            (int)(Picker->NewColor.g * 255.0f),
+            (int)(Picker->NewColor.b * 255.0f));
+        if (ImGui::InputText("Hex", hex_color, sizeof(hex_color) / sizeof(hex_color[0]), ImGuiInputTextFlags_CharsHexadecimal)) {
```
Comment (ApoorvaJ) [inline src/core/picker.cpp]: Curly bracket on it's own line, please. :)

if (...)
{
    ...
}

File: src/core/picker.cpp [hunk 6]
```diff
@@ -82,16 +82,38 @@ void Picker::Show(PickerInfo* Picker, Color* Colors, uint32 BlankTexture) // TOD
...
+            int r = 0, g = 0, b = 0;
+            switch (strlen(hex_color)) {
```
Comment (ApoorvaJ) [inline src/core/picker.cpp]: Same here. Bracket on next line.

**Comment by ApoorvaJ** [comment]
Thanks for your work!

I have made a few review notes on the pull request. Nothing too big, just slight changes to get you used to the code style.

Some comments have gone to "an older commit", but I'm not sure why. You can still view them, though. They are still valid comments.

Please have a look, and I'll merge it in. :)

**Comment by GreenLightning** [comment]
I amended the commit to fix the things you pointed out. Thanks for pointing stuff out :)

**Comment by ApoorvaJ** [comment]
Thanks for updating the pull request. Good work! :)

**Pull request merged**